scripts/resolve_container_path.py

```diff
@@ -19,24 +29,47 @@
 
 def resolve_container_path(container_path):
     container_path = os.path.realpath(container_path)
+    mountinfo_path = '/proc/self/mountinfo'
     try:
-        with open('/proc/self/cgroup') as f:
-            # Entries in /proc/self/cgroup look like this (note the nesting):
-            # 11:name=systemd:/docker/82c1bd2…23b5bcf/docker/6547bce…60ca5a7
-            prefix, container_id = next(f).strip().split(':')[2].split('/')[-2:]
+        with open(mountinfo_path) as f:
+            log.info('Found %s', mountinfo_path)
+            # Entries of interest in /proc/self/mountinfo look as follows:
+            # 752 744 259:2 /docker/containers/dc61d9…/…
+            # dc61d9… is the container ID
+            prefix = '/docker/containers/'
+            container_ids = set()
+            for line in f:
+                assert line.endswith('\n'), line
+                parts = line.strip().split(' ')
+                assert len(parts) > 9, parts
+                root = parts[3]
+                for expected_prefix in (f'/var/lib{prefix}', prefix):
+                    if root.startswith(expected_prefix):
+                        log.info('Extracting container ID from %s', root)
+                        id, *rest = root.removeprefix(expected_prefix).split('/')
+                        assert len(id) == 64 and int(id, base=16), id
```

Comment:
This would fire for 64 zeros, which we have no reason to regard as invalid. Also, it allows lower- and uppercase hexadecimal digits, but I have only ever seen lowercase in the wild. Using a regex is probably better in this case.